spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/execution/TruncateArcticTableExec.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.netease.arctic.spark.sql.execution
+
+import com.netease.arctic.op.OverwriteBaseFiles
+import com.netease.arctic.spark.table.{ArcticIcebergSparkTable, ArcticSparkTable}
+import com.netease.arctic.utils.TablePropertyUtil
+import org.apache.iceberg.expressions.Expressions
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.catalog.CatalogTypes.TablePartitionSpec
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.connector.catalog.Table
+import org.apache.spark.sql.execution.datasources.v2.V2CommandExec
+
+case class TruncateArcticTableExec(table: Table,
+                                   partitionSpec: Option[TablePartitionSpec]) extends V2CommandExec {
+  override protected def run(): Seq[InternalRow] = {
+    if (partitionSpec.isDefined) {
+      throw new UnsupportedOperationException("Not support truncate partition temporarily")
+    }
+    table match {
+      case arctic: ArcticSparkTable =>
+        if (arctic.table().isKeyedTable) {
+          val overwriteBaseFiles: OverwriteBaseFiles = arctic.table().asKeyedTable().newOverwriteBaseFiles()
+          overwriteBaseFiles.overwriteByRowFilter(Expressions.alwaysTrue())
+          overwriteBaseFiles.withTransactionIdForChangedPartition(
+            TablePropertyUtil.allocateTransactionId(arctic.table().asKeyedTable()))
+          overwriteBaseFiles.commit()
+        } else {
+          val overwriteFiles = arctic.table().asUnkeyedTable().newOverwrite()
+          overwriteFiles.overwriteByRowFilter(Expressions.alwaysTrue())
+          overwriteFiles.commit()
+        }
+      case arctic: ArcticIcebergSparkTable =>
+        val overwriteFiles = arctic.table().newOverwrite()
+        overwriteFiles.overwriteByRowFilter(Expressions.alwaysTrue())
+        overwriteFiles.commit()
```

Comment:
Still has problems, for native hive table, if not matched here, sql will execute failed.